idemp-base/src/org/bandahealth/idempiere/base/modelevent/BusinessPartnerModelEvent.java

```diff
@@ -143,9 +147,18 @@ private void beforeSaveRequest(MBPartner_BH businessPartner) {
 	}
 
 	private void translateToMaskedFields(MBPartner_BH businessPartner) {
-
 		businessPartner.setIsCustomer(businessPartner.isBH_IsPatient());
 		businessPartner.set_ValueOfColumn(MBPartner_BH.COLUMNNAME_BH_ApproximateYears, null);
+		// Make sure the birthday doesn't have a time
+		if (businessPartner.getBH_Birthday() != null) {
+			Calendar calendar = GregorianCalendar.getInstance();
+			calendar.setTimeInMillis(businessPartner.getBH_Birthday().getTime());
```

Comment:
It was added because I was getting into an infinite loop on my machine - not sure why this is the first time seeing it. Anyways, this should probably be pulled out to it's own review.

Regarding the time, one of the birthday columns (I'd have to check which) doesn't track time, so having time resulted in the Timestamps being different on each object. I agree with the time, especially for babies, so maybe the right solution is to add the time to whichever column it is in the DB.

Reply:
I've undone this change, though can't test without the [go-2405](https://issues.bandahealth.org/issue/go-2405/Update-how-value-column-used-on-products) fix. Will see if the problem re-appears afterwards and I'll create a separate issue to address that problem when/if it occurs.